ui/app/scripts/controllers/newDeploymentCtrl.js

```diff
@@ -476,10 +732,15 @@ angular.module('apollo')
 		    if (environmentId !== loadedGroupsEnvironmentId || serviceId !== loadedGroupsServiceId) {
 		        loadedGroupsEnvironmentId = environmentId;
 		        loadedGroupsServiceId = serviceId;
-                apolloApiService.getGroupsPerServiceAndEnvironment(environmentId, serviceId).then(function (response) {
-                    $scope.possibleGroups = response.data;
+                return apolloApiService.getGroupsPerServiceAndEnvironment(environmentId, serviceId).then(function (response) {
+                    response.data.forEach(function(group) {
+                        $scope.possibleGroups.push(group);
+                        $scope.possibleGroupsIds.push(group.id);
```

Comment:
This is not being used. please remove this from there and the declaration

Reply:
it's in use. without it, it won't know the possible groups of the selected environments.

Reply:
Where? I searched your entire PR. Its just being declared, and assigned here. No one is looking on that on the JS or HTML side